### DISCUSSION TITLE
ONVIF Client Test Tool

### DISCUSSION THREAD

**Discussion by RM-Arohi** [onvif/testspecs#25] [Q&A] [answered]
Hello,

I'm trying the tool on our vms software to get the details, however bit puzzled here with the Feature List XML file which I do not have. Can anyone please help on how can it be generated? 
Also Is Feature List XML file mandatory? Can't I run the test without it ? I've tried using the tool with the feature list xml file, however none of the feature which I tried passed. attached is the file for reference.
[OnvifClientTool.docx](https://github.com/onvif/testspecs/files/8069439/OnvifClientTool.docx)

Can anyone help here how can I run the tool successfully ? Thanks in advance.

Best Regards,
RM-Arohi

**Comment by madhurao68** [comment, marked as answer]
Please refer the section on Configuration tab in the Client test tool Help for help on configuring the traces and feature list xml file. For a successful conformance you will need to parse the network trace files to be used in conformance testing and subsequently upload the feature list xml file for the identified devices.  Feature list for a specific device can be downloaded from the ONVIF conformant products page at: https://www.onvif.org/conformant-products/

**Comment by RM-Arohi** [comment]
Hello Madhu,

Thanks for your response. Below are the steps I've done currently 
refering the help file... please let me know if its correct.

1. I'm using Server (our vms) and Client (thick and thin client) 
architecture

2. For the trace file, I'm using Wireshark on my client machine. Even 
tried generating the logs on the server using Wireshark.

3. Added the trace/log file i.e. *.pcapng file in the Configuration Tab 
and ran the option of 'Start Parsing'.

4. The IPs are shown on the right side under 'Parsing Results'. Now, as 
my vms doesn't have feature list xml file I goto the next step.

5. Under Configuration Tab  >> Client Settings, I select the options 
which I need.

6. Under Configuration Tab  >> Device Features, I select my vms server's 
IP which is listed.

7. Then under Diagnostics tab I start the test using 'Run Conformance 
Test', here I get a waring saying '/Some devices do not have Feature 
list //xml attached. Conformance may not be possible/'. As I do not have 
Feature list xml file of our vms, by clicking Ok I'm going ahead with 
the process.

8. Once Conformance Test' is done 100% under Output I see maximum of the 
features shows as 'Not Detected' and on the left side maximum of the 
options are in Red. The featues which I've selected is configured in my 
vms and the same I'm accessing using client machine during the Wireshark 
logs are getting generated. Even though I've configured Recording (3 
cams). Motion detection (3 cams), using PTZ funtions of 1 camera, etc., 
I'm NOT even getting Profile S basic featuers like User Token, Streaming 
options in Green.

As per my understanding once the above point is successful I will be 
able to Generate the feature list xml file from Conformance tab. Since 
no features are detected how can I provide the xml file before under 
Parsing results ? Also, hope you have seen my attached file which I'd 
sent earlier in my first question.

Hope the procedure is fine and I'm correct ? Kindly let me know if I'm 
missing anything here.

Thanks, RM-Arohi


Madhu Rao wrote:

**Comment by madhurao68** [comment]
Hello,
For passing conformance for the chosen profile you will need to upload the Feature List xml files of the devices supporting these features. Please note that it is mandatory to upload feature list files of more than 2 manufacturers for each feature of the supported profiles.  Feature list files for desired devices can be downloaded from the Conformant Products page (https://www.onvif.org/conformant-products/). Also refer Conformance process  workflow at https://www.onvif.org/wp-content/uploads/2019/03/ONVIF_Member-Conformance-Process-Overview.pdf.
Hope this helps.

**Comment by RM-Arohi** [comment]
Hello,

Yes, I've done that already and I'm using 3 device of different brand.
However attached is the error I'm getting, so don't know why the Profile S
mandatory features are not passed.
Any idea ?

Thanks, RM-Arohi

**Reply by RM-Arohi** [reply]
Error screenshot

![OnvifTool_ProfileError](https://user-images.githubusercontent.com/99736933/155092672-292e63d7-78b6-44fa-8209-4783db7f8493.jpeg)

**Comment by madhurao68** [comment]
Hello,
For every identified unit listed (devices) in the Parsing Results tab (this list gets populated by the tool based on the traces you add in the Configuration tab)  you must upload the applicable feature list xml file. So for example, if the list shows D1 as the device from a manufacturer M1 the feature list for that specific device D1 must be uploaded. In short you must choose the correct feature list for the devices you need to test.  
Please also ensure that you have the traces for all the mandatory features before proceeding with the above step.  The result of the conditional features has no bearing on the conformance. 
Hope this helps!

**Comment by RM-Arohi** [comment]
Hello,

Yes, I've all the correct feature list xml files for the devices I'm 
using which I've downlaoded from the Onvif Conformance Produt page.

Even though the mandatory features are enabled and working during the 
traces I'm getting the errors... so a bit confused whether anything I'm 
missing or not.

Thanks, Roopesh


Madhu Rao wrote:

**Reply by madhurao68** [reply]
Hi Roopesh,
Will you be able to send the traces with the device models for me to evaluate ? It is not possible to determine the exact cause of the failure without this information. You can mail me at madhu.rao@developer.onvif.org
Thanks!

**Comment by madhurao68** [comment]
Hi,

Thanks for sending the traces and feature lists. 

If you check the status of the traces in the Configuration tab you will see traces with error messages (move the cursor over the exclamation mark and you can see the message).  It is clear that the traces do not have the conversations for the mandatory features (refer the conformance output which shows all the mandatory features in fail status). You will need to add traces with the conversations between the client and device for these missing features for the conformance to succeed.

**Comment by RM-Arohi** [comment]
Hi,

Yes saw that, however we have even enabled the cameras using udp 
protocol still getting the error as attached.

Thanks, Roopesh


Madhu Rao wrote:

**Comment by SergeyBogdanov** [comment]
Hi,

From the traces I see they do not contain expected ONVIF commands. So, Client Test Tool can not prove the client supports some ONVIF operations (called features in the test tool) mandatory for Profile S.
Basically, the Client Test Tool verifies feature support in the following way: It parses the traces. In each test case the tool goes through parsed traces and looks for specific set of requests/responses for ONVIF commands. It may be SOAP commands or RTSP commands. Once the test reports it found all necessary commands (according to test's logic) it is marked as passed.

In this particular case I would collect additional traces for missed commands. Say, to prove support for ‘Capabilities’ feature, it needs: 
* Start Wireshark to capture trace
* From client software/hardware initiate request for device’s capabilities for a camera
* In Wireshark stop and save the trace
* Upload the trace to CTT and run conformance test
* After the test is completed, the tree item ‘Profile S->Mandatory->Capabilities->Passed with Devices’ should show ‘1’. The color of one of test cases: CAPABILITIES-1 GET SERVICES or CAPABILITIES-2 GET CAPABILITIES should become green. That means the conversation is detected and the test case is passed.
* Repeat above steps for two other devices
* After that, the feature ‘Capabilities’ should be passed

It is necessary to collect traces the same way for all operations required for Profile S to pass Profile S conformance.

You may find more details on which commands is expected in each test case from test specification. On the 'Diagnostics' tab, if you select a test case, you can find the test specification for this test case in the right side of form, on the 'Specification' tab.

Hope this helps,
Sergey Bogdanov

**Comment by RM-Arohi** [comment]
Hi Sergey Bogdanov,

Thanks for the information. We are trying the same however we are still
facing issues.
Will do some few test runs and update here.

Thanks, RM-Arohi

**Comment by RM-Arohi** [comment]
Hi All,

Thanks for your time and support.

Got the success with onvif conformance test... reason was the firmware 
version of the cameras which I was using.

Best Regards,

RM-Arohi


Roopesh Mistry wrote:
> Hi Sergey Bogdanov,
>
> Thanks for the information. We are trying the same however we are 
> still facing issues.
> Will do some few test runs and update here.
>
> Thanks, RM-Arohi
>
> On Wed, 2 Mar 2022 at 16:12, SergeyBogdanov ***@***.***
